src/main/perl/Application.pm

```diff
@@ -121,20 +121,62 @@ sub username
 
 =pod
 
+=item option_exists($opt): boolean
+
+Returns true if the option exists, false otherwhise. Option can be
+defined either in the application configuration file or on the
+command line (based on C<AppConfig> module).
+
+=cut
+
+# Check if a configuration option exists
+sub option_exists
+{
+    my ($self, $option) = @_;
+    return $self->{CONFIG}->_exists($option);
+}
+
+=pod
+
 =item option($opt): scalar|undef
 
 Returns the option value coming from the command line and/or
 configuration file. Scalar can be a string, or a reference to a hash
 or an array containing the option's value. option() is a wrapper
-on top of AppConfig->get($opt).
+on top of AppConfig->get($opt). If the option doesn't exist, returns
+C<undef>.
 
 =cut
 
 sub option
 {
     my ($self,$opt) = @_;
 
-    return $self->{'CONFIG'}->get($opt);
+    if ( $self->option_exists($opt) ) {
+        return $self->{'CONFIG'}->get($opt);
+    }
+
+    return;
+}
+
+=pod
+
+=item set_option($opt, $val): undef
+
+Defines an option and sets its value. If the option was previously
+defined, its value is overwritten. This is a wrapper over C<AppConfig>
+methods to hide the internal implementation of a C<CAF::Application>.
+
+=cut
+
+sub set_option
+{
+    my ($self,$opt,$val) = @_;
+
+    $self->{'CONFIG'}->define($opt);
+    $self->{'CONFIG'}->set($opt, $val);
+
+    return;
```

Comment:
maybe return `$val`? or `SUCCESS`? altough nothing can go wrong, it's odd that we return undef when something is ok